### PR TITLE
[RLlib] DDPG Training iteration fn & Replay Buffer API

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

Conversion to Training Iteration Fn for DDPG.
Also conversion to replay buffer api.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/ddpg/ddpg.py [hunk 1]
```diff
@@ -84,7 +84,7 @@
         "scale_timesteps": 10000,
     },
     # Number of env steps to optimize for before returning
```
Comment (smorad) [inline rllib/agents/ddpg/ddpg.py:86]: Fix comment

File: rllib/agents/ddpg/ddpg.py [hunk 2]
```diff
@@ -94,8 +94,15 @@
     # each worker will have a replay buffer of this size.
     "buffer_size": DEPRECATED_VALUE,
     "replay_buffer_config": {
-        "type": "MultiAgentReplayBuffer",
+        "_enable_replay_buffer_api": True,
+        "type": "MultiAgentPrioritizedReplayBuffer",
         "capacity": 50000,
+        # Alpha parameter for prioritized replay buffer.
+        "prioritized_replay_alpha": 0.6,
```
Comment (smorad) [inline rllib/agents/ddpg/ddpg.py:101]: DDPG now with PER :D

Comment (ArturNiederfahrenhorst) [reply rllib/agents/ddpg/ddpg.py:101]: It's had PER all along!

File: rllib/BUILD [hunk 3]
```diff
@@ -228,25 +228,25 @@ py_test(
...
-# py_test(
-#    name = "learning_tests_pendulum_ddpg",
-#    main = "tests/run_regression_tests.py",
-#    tags = ["team:ml", "learning_tests", "learning_tests_pendulum", "learning_tests_continuous"],
-#    size = "large",
-#    srcs = ["tests/run_regression_tests.py"],
-#    data = glob(["tuned_examples/ddpg/pendulum-ddpg.yaml"]),
-#    args = ["--yaml-dir=tuned_examples/ddpg"]
-# )
+py_test(
```
Comment (sven1977) [inline rllib/BUILD:231]: Noice!! :)

File: rllib/agents/ddpg/ddpg.py [hunk 4]
```diff
@@ -178,7 +177,7 @@
     # If True, the execution plan API will not be used. Instead,
     # a Trainer's `training_iteration` method will be called as-is each
     # training iteration.
```
Comment (sven1977) [inline rllib/agents/ddpg/ddpg.py:179]:
```suggestion
```

File: rllib/agents/ddpg/ddpg.py [hunk 5]
```diff
@@ -178,7 +177,7 @@
     # If True, the execution plan API will not be used. Instead,
     # a Trainer's `training_iteration` method will be called as-is each
```
Comment (sven1977) [inline rllib/agents/ddpg/ddpg.py:178]:
```suggestion
```

File: rllib/agents/ddpg/ddpg.py [hunk 6]
```diff
@@ -178,7 +177,7 @@
     # If True, the execution plan API will not be used. Instead,
```
Comment (sven1977) [inline rllib/agents/ddpg/ddpg.py:177]:
```suggestion
```

File: rllib/agents/ddpg/ddpg.py [hunk 7]
```diff
@@ -178,7 +177,7 @@
...
     # a Trainer's `training_iteration` method will be called as-is each
     # training iteration.
-    "_disable_execution_plan_api": False,
+    "_disable_execution_plan_api": True,
```
Comment (sven1977) [inline rllib/agents/ddpg/ddpg.py:180]:
```suggestion
```

File: rllib/tuned_examples/ddpg/invertedpendulum-td3.yaml [hunk 8]
```diff
@@ -21,5 +21,5 @@ invertedpendulum-td3:
             random_timesteps: 1000
 
         # === Evaluation ===
-        evaluation_interval: 1
+        evaluation_interval: 10
         evaluation_num_episodes: 5
```
Comment (sven1977) [inline rllib/tuned_examples/ddpg/invertedpendulum-td3.yaml:25]:
```suggestion
        evaluation_duration: 5
```

File: rllib/tuned_examples/ddpg/mujoco-td3.yaml [hunk 9]
```diff
@@ -23,5 +23,5 @@ mujoco-td3:
             random_timesteps: 10000
 
         # === Evaluation ===
-        evaluation_interval: 5
+        evaluation_interval: 10
         evaluation_num_episodes: 10
```
Comment (sven1977) [inline rllib/tuned_examples/ddpg/mujoco-td3.yaml:27]:
```suggestion
        evaluation_duration: 10
```

**Review by sven1977 — APPROVED** [review]
Nice! Thanks for crossing out yet another two alsos from our `training_iteration` TODO list. :)

**Pull request merged**